tools/lintdiff.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
+"""
+Drop in replacement for golangci-lint that runs it only on changed packages.
+
+Changes are calculated as diff against main by default, use --ref or -H/--head to change this.
+"""
+
+import os
+import sys
+import argparse
+import subprocess
+
+
+def parse_lines(cmd):
+    # print("+ " + " ".join(cmd), file=sys.stderr, flush=True)
+    result = subprocess.run(cmd, stdout=subprocess.PIPE, encoding="utf-8", check=True)
+    return [x.strip() for x in result.stdout.split("\n") if x.strip()]
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--ref", default="main", help="Reference to calculate diff against.")
+    parser.add_argument("-H", "--head", action="store_true", help="Shortcut for '--ref HEAD' - test uncommitted changes only")
+    parser.add_argument("args", nargs=argparse.REMAINDER, help="golangci-lint command and options")
+    args = parser.parse_args()
+
+    if not args.args:
+        args.args = ["run"]
+
+    if args.head:
+        args.ref = "HEAD"
+
+    gitroot = parse_lines(["git", "rev-parse", "--show-toplevel"])[0]
+    os.chdir(gitroot)
```

Comment:
Why the need to chdir? Relative package paths work just the same.

Reply:
We get paths from git, which prints them relative to repo root regardless of where it is run:

```
~/work/cli-main/libs/git % git diff --name-only main -- .
libs/git/config.go
```